server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -303,15 +302,20 @@ private WatchKey enableSettingsWatcher(WatchKey previousKey, Path settingsDir) t
     CountDownLatch processFileSettings(Path path, Consumer<Exception> errorHandler) throws IOException {
         CountDownLatch waitForCompletion = new CountDownLatch(1);
         logger.info("processing path [{}] for [{}]", path, NAMESPACE);
-        try (var json = new BufferedInputStream(Files.newInputStream(path))) {
-            try (XContentParser parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, json)) {
-                stateService.process(NAMESPACE, parser, (e) -> {
+        try (
```

Comment:
This is certainly more explicit but I don't think this is actually anymore _correct_. Closing the `BufferedInputStream` should close the delegate `FileInputStream` implicitly but as you say, it's worth ruling things out.

Reply:
Agreed.